src/windows/player.c

```diff
@@ -264,6 +276,54 @@ static void window_player_overview_show_group_dropdown(rct_window *w, rct_widget
 	dropdown_set_checked(network_get_group_index(network_get_player_group(player)), true);
 }
 
+static void window_player_overview_show_kick_dropdown(rct_window *w, rct_widget *widget)
+{
+	rct_widget *dropdownWidget;
+	int numItems, i;
+	int player = network_get_player_index((uint8)w->number);
+	if (player == -1) {
+		return;
+	}
+
+	dropdownWidget = widget;
+
+	numItems = KickReasonCount;
+	for (i = 0; i < numItems; i++) {
+		gDropdownItemsFormat[i] = KickReasonStringIds[i];
+		gDropdownItemsArgs[i] = 0;
+	}
+	window_dropdown_show_text(
+		w->x + dropdownWidget->left,
+		w->y + dropdownWidget->top,
+		dropdownWidget->bottom - dropdownWidget->top + 1,
+		w->colours[1],
+		0,
+		numItems
+	);
+	gDropdownDefaultIndex = 0;
+	gDropdownHighlightedIndex = 0;
+}
+
+static void window_player_overview_kick_player(rct_window* w, int strReasonID, const char* reason)
+{
+	uint8 playerid = (uint8)w->number;
+	int player = network_get_player_index(playerid);
+	if (player == -1) {
+		return;
+	}
+	char msg[NETWORK_DISCONNECT_KICK_REASON_MAX_SIZE];
+	if (strReasonID == STR_KICK_REASON_OTHER_REASON && (reason && strcmp(reason, "") != 0)) {
+		strncpy(msg, reason, NETWORK_DISCONNECT_KICK_REASON_MAX_SIZE);
+	}
+	else if (strReasonID == STR_KICK_REASON_NO_REASON || strReasonID == STR_KICK_REASON_OTHER_REASON) {
+		format_string(msg, STR_MULTIPLAYER_KICKED_REASON, NULL);
+	}
+	else {
+		format_string(msg, strReasonID, NULL);
+	}
+	network_kick_player(playerid, msg);
```

Comment:
Yes it would, but as the current network system works, that is not yet available. The current way the chat system works for notifications is it localises the admins language and then sends that text to clients. To show a localised chat message would mean to have to rewrite the chat functionality entirely to show a client-side string id, which would have to be a seperate pull request and feature.